src/workerd/jsg/ser.c++

```diff
@@ -110,6 +227,52 @@ v8::Maybe<bool> Serializer::WriteHostObject(v8::Isolate* isolate, v8::Local<v8::
   try {
     jsg::Lock& js = jsg::Lock::from(isolate);
 
+    if (object->IsNativeError()) {
+      auto nameStr = js.str("name"_kj);
+      auto messageStr = js.str("message"_kj);
+
+      // Get the standard name, message, stack, cause, error, errors properties from
+      // the error object.
+      writeRawUint32(SERIALIZATION_TAG_NATIVE_ERROR);
+
+      // A mix of ad-hoc and regular serialization. We first serialize
+      // the error tag, which is an enum that identifies the type of error
+      // for faster/easier deserialization. Then we serialize the message which
+      // usually come from the prototype. Then we grab the own properties,
+      // serializing the number of them followed by each name and value in
+      // sequence.
+
+      jsg::JsObject errorObj(object);
+      auto name = errorObj.get(js, nameStr);
+      auto tag = getErrorTagFromName(js, name);
+      writeRawUint32(static_cast<uint32_t>(tag));
+      // We only write the name if it is not one of the known error types.
+      if (tag == ErrorTag::UNKNOWN) {
+        write(js, name);
+      }
+
+      write(js, errorObj.get(js, messageStr));
+
+      auto names = errorObj.getPropertyNames(js, KeyCollectionFilter::OWN_ONLY,
```

Comment:
```suggestion
      // This will cover not only the extension properties we add via tunneled exceptions,
      // along with any additional own properties the user may attach to the error object,
      // it also ensures that the standard `cause`, `error`, `errors`, and `suppressed`
      // properties that are defined by the language standard are also captured and
      // serialized. Later, when v8 finally gets around to supporting these by default
      // we can likely filter them out.
      auto names = errorObj.getPropertyNames(js, KeyCollectionFilter::OWN_ONLY,
```